callback.php

```diff
@@ -100,13 +100,21 @@ function insertTXIDIntoPaymentDetails($presta_order, $txid, $blockonomics_order)
 {
     $paid_ratio = $blockonomics_order['bits_payed'] / $blockonomics_order['bits'];
     $amount = round($paid_ratio * $blockonomics_order['value'], 2);
+
+    // Get invoice and save the address used in the successful payment as an invoice note
+    $presta_order = new Order($blockonomics_order['id_order']);
+    $invoice = $presta_order->getInvoicesCollection()[0];
+    $invoice_note = Tools::strtoupper($blockonomics_order['crypto']) . " Address: " . $blockonomics_order['addr'];
+    $invoice->note = $invoice_note;
+    $invoice->save();
     
     $payments = $presta_order->getOrderPayments();
     if (!$payments) {
-        $payment_method = 'Bitcoin - Blockonomics';
+        $payment_method = "Blockonomics - " . Tools::strtoupper($blockonomics_order['crypto']);
         $presta_order->addOrderPayment($amount, $payment_method, $txid);
-    } elseif ($payments[0]->payment_method = 'Bitcoin - Blockonomics' && !$payments[0]->transaction_id) {
+    } elseif (!$payments[0]->transaction_id) {
```

Comment:
Removing the payment_method check may lead to issues when using multiple payment methods or cryptos.
It will be best to still verify the payment method matches and the crypto match:
`$payments[0]->payment_method == "Blockonomics - " . Tools::strtoupper($blockonomics_order['crypto'])`

Reply:
It seems like setting `$o->setCurrentState(Configuration::get('PS_OS_PAYMENT'));` creates a payment and uses the module name as payment method, that is `Bitcoin - Blockonomics`. The image shows the result of the callback being called once.
![image](https://user-images.githubusercontent.com/36883813/122014727-68e09980-cdfa-11eb-849b-96fb74abb2f7.png)

The only difference is that I put `insertTXIDIntoPaymentDetails();` before changing the order status. That resulted in us first adding a payment and then Presta adding a payment.

I moved back `insertTXIDIntoPaymentDetails();` to its original position, but instead of checking for 
`$payments[0]->payment_method == "Blockonomics - " . Tools::strtoupper($blockonomics_order['crypto'])` it checks for `$payments[0]->payment_method == "Bitcoin - Blockonomics"`.